# # Part1. 인하대 후문 외식업소 데이터 정리
- 데이터 출처 : 소상공인시장진흥공단_상가(상권)정보(https://www.data.go.kr/data/15083033/fileData.do)
- 전국 데이터 중 인하대 후문 상권 데이터를 받아옴
- 가장 중요한 피처는 상호명, 업종, 위치(주소)
- 불필요한 데이터를 없애고 업종을 통일해서 분류시키는 작업이 필요

In [1]:
import pandas as pd

### 1) 데이터 확인

In [2]:
inha_restaurant = pd.read_csv("C:/Users/sj123/Downloads/inha_restaurant.csv")
inha_restaurant

field1 field2      field3   field4                  field5      field6  \
0         1  일반음식점  07_24_04_P  3510500  3510500-101-2023-00285  2023-07-26   
1         2  일반음식점  07_24_04_P  3510500  3510500-101-2023-00305  2023-08-07   
2         3  일반음식점  07_24_04_P  3510500  3510500-101-2023-00333  2023-09-04   
3         4  일반음식점  07_24_04_P  3510500  3510500-101-2018-00412  2018-06-28   
4         5  일반음식점  07_24_04_P  3510500  3510500-101-2019-00164  2019-05-01   
..      ...    ...         ...      ...                     ...         ...   
213     214  일반음식점  07_24_04_P  3510500  3510500-101-2023-00389  2023-10-05   
214     215  일반음식점  07_24_04_P  3510500  3510500-101-2023-00424  2023-11-03   
215     216  일반음식점  07_24_04_P  3510500  3510500-101-2024-00433  2024-12-09   
216     217  일반음식점  07_24_04_P  3510500  3510500-101-2024-00421  2024-12-03   
217     218  일반음식점  07_24_04_P  3510500  3510500-101-2024-00443  2024-12-16   

     field7  field8 field9  field10  ... field33 field34  field35  field36  \
0       NaN       1  영업/정상        1  ...     0.0     0.0      0.0      NaN   
1       NaN       1  영업/정상        1  ...     0.0     0.0      0.0      NaN   
2       NaN       1  영업/정상        1  ...     0.0     0.0      0.0      NaN   
3       NaN       1  영업/정상        1  ...     NaN     NaN      NaN      NaN   
4       NaN       1  영업/정상        1  ...     NaN     NaN      NaN      NaN   
..      ...     ...    ...      ...  ...     ...     ...      ...      ...   
213     NaN       1  영업/정상        1  ...     0.0     0.0      0.0      NaN   
214     NaN       1  영업/정상        1  ...     0.0     0.0      0.0      NaN   
215     NaN       1  영업/정상        1  ...     0.0     0.0      0.0      NaN   
216     NaN       1  영업/정상        1  ...     0.0     0.0      0.0      NaN   
217     NaN       1  영업/정상        1  ...     0.0     0.0      0.0      NaN   

    field37 field38  field39 field40 _GC_TYPE                     _CLEANADDR  
0       0.0     0.0        N   90.24        정      인천광역시 미추홀구 인하로 73-1 (용현동)  
1       0.0     0.0        N   80.50        정  인천광역시 미추홀구 인하로67번길 6-23 (용현동)  
2       0.0     0.0        N   30.00        정       인천광역시 미추홀구 인하로 152 (용현동)  
3       NaN     NaN        N   40.88        정    인천광역시 미추홀구 인하로77번길 26 (용현동)  
4       NaN     NaN        N  170.00        정      인천광역시 미추홀구 인하로 55-1 (용현동)  
..      ...     ...      ...     ...      ...                            ...  
213     0.0     0.0        N   40.00        정   인천광역시 미추홀구 인하로147번길 28 (용현동)  
214     0.0     0.0        N   49.84        정       인천광역시 미추홀구 인하로 117 (용현동)  
215     0.0     0.0        N   82.56        정  인천광역시 미추홀구 인하로67번길 6-31 (용현동)  
216     0.0     0.0        N  100.80        정  인천광역시 미추홀구 인하로67번길 6-31 (용현동)  
217     0.0     0.0        N   66.00        정    인천광역시 미추홀구 인하로77번길 18 (용현동)  

[218 rows x 42 columns]

In [3]:
inha_restaurant.columns

Index(['field1', 'field2', 'field3', 'field4', 'field5', 'field6', 'field7',
       'field8', 'field9', 'field10', 'field11', 'field12', 'field13',
       'field14', 'field15', 'field16', 'field17', 'field18', 'field19',
       'field20', 'field21', 'field22', 'field23', 'field24', 'field25',
       'field26', 'field27', 'field28', 'field29', 'field30', 'field31',
       'field32', 'field33', 'field34', 'field35', 'field36', 'field37',
       'field38', 'field39', 'field40', '_GC_TYPE', '_CLEANADDR'],
      dtype='object')

### 2) 불필요한 칼럼 모두 제거

In [4]:
inha_restaurant.drop(['field1', 'field3', 'field4', 'field5', 'field6',
                      'field7', 'field8', 'field10', 'field11', 'field12',
                      'field13', 'field14', 'field17', 'field19',
                      'field20', 'field21', 'field23', 'field24',
                      'field25', 'field26', 'field27', 'field28', 
                      'field29', 'field30','field31', 'field32', 
                      'field33', 'field34', 'field35', 'field36', 
                      'field37', 'field38', 'field39', 'field40','_GC_TYPE', '_CLEANADDR'], axis=1, inplace=True)
inha_restaurant.head()

field2 field9                       field15  \
0  일반음식점  영업/정상          인천광역시 미추홀구 용현동 191-4   
1  일반음식점  영업/정상         인천광역시 미추홀구 용현동 191-94   
2  일반음식점  영업/정상  인천광역시 미추홀구 용현동 39-26 1충동 좌측호   
3  일반음식점  영업/정상       인천광역시 미추홀구 용현동 181-17번지   
4  일반음식점  영업/정상          인천광역시 미추홀구 용현동 199-4   

                             field16     field18 field22  
0  인천광역시 미추홀구 인하로 73-1, 1층 일부호 (용현동)  백소정 인하대후문점      일식  
1  인천광역시 미추홀구 인하로67번길 6-23, 2층 (용현동)        하하돈면      기타  
2   인천광역시 미추홀구 인하로 152, 1층 좌측호 (용현동)     밥줄까 술주까      한식  
3        인천광역시 미추홀구 인하로77번길 26 (용현동)       셰프하우스      한식  
4          인천광역시 미추홀구 인하로 55-1 (용현동)        육쌈냉면      분식

In [5]:
inha_restaurant.columns

Index(['field2', 'field9', 'field15', 'field16', 'field18', 'field22'], dtype='object')

In [6]:
inha_restaurant.shape

(218, 6)

### 3) 칼럼명 바꿔주기

In [7]:
inha_restaurant.columns = ['대분류', '영업여부', '구주소', '도로명 주소', '상호명', '업종']

In [8]:
inha_restaurant.head()

대분류   영업여부                           구주소  \
0  일반음식점  영업/정상          인천광역시 미추홀구 용현동 191-4   
1  일반음식점  영업/정상         인천광역시 미추홀구 용현동 191-94   
2  일반음식점  영업/정상  인천광역시 미추홀구 용현동 39-26 1충동 좌측호   
3  일반음식점  영업/정상       인천광역시 미추홀구 용현동 181-17번지   
4  일반음식점  영업/정상          인천광역시 미추홀구 용현동 199-4   

                              도로명 주소         상호명  업종  
0  인천광역시 미추홀구 인하로 73-1, 1층 일부호 (용현동)  백소정 인하대후문점  일식  
1  인천광역시 미추홀구 인하로67번길 6-23, 2층 (용현동)        하하돈면  기타  
2   인천광역시 미추홀구 인하로 152, 1층 좌측호 (용현동)     밥줄까 술주까  한식  
3        인천광역시 미추홀구 인하로77번길 26 (용현동)       셰프하우스  한식  
4          인천광역시 미추홀구 인하로 55-1 (용현동)        육쌈냉면  분식

### 4) 순서 재정비
- 인하대 후문 중심으로 218곳 필터링
- 주요 필드만 정제

In [9]:
order = ['대분류', '상호명', '업종', '영업여부', '구주소', '도로명 주소']
inha_restaurant = inha_restaurant[order]

In [10]:
inha_restaurant.head(3)

대분류         상호명  업종   영업여부                           구주소  \
0  일반음식점  백소정 인하대후문점  일식  영업/정상          인천광역시 미추홀구 용현동 191-4   
1  일반음식점        하하돈면  기타  영업/정상         인천광역시 미추홀구 용현동 191-94   
2  일반음식점     밥줄까 술주까  한식  영업/정상  인천광역시 미추홀구 용현동 39-26 1충동 좌측호   

                              도로명 주소  
0  인천광역시 미추홀구 인하로 73-1, 1층 일부호 (용현동)  
1  인천광역시 미추홀구 인하로67번길 6-23, 2층 (용현동)  
2   인천광역시 미추홀구 인하로 152, 1층 좌측호 (용현동)

In [13]:
inha_restaurant['업종'] = inha_restaurant['업종'].str.strip()
categories = sorted(inha_restaurant['업종'].unique())

categories

['감성주점',
 '경양식',
 '기타',
 '분식',
 '뷔페식',
 '식육(숯불구이)',
 '외국음식전문점(인도,태국등)',
 '일식',
 '정종/대포집/소주방',
 '중국식',
 '탕류(보신용)',
 '통닭(치킨)',
 '패스트푸드',
 '한식',
 '호프/통닭',
 '횟집']

### 5) 저장

In [14]:
inha_restaurant.to_csv('inha_restaurant_FE.csv', index=False, encoding='utf-8-sig')